pkg/controller/registry/resolver/constraints/cel.go

```diff
@@ -0,0 +1,127 @@
+package constraints
```

Comment:
Why is this code duplicated in both API and OLM? Shouldn't OLM just import from API?

Reply:
The PR on api repo was created later after I was adding some validation on registry repo and realized there were some shared codes between OLM and registry that can be consolidated in api repo.
I'm keeping the code here for now so that all of tests can be run properly while waiting for the api PR to merge. Even if we merge this PR first, I can do a subsequent PR to vendor new codes from api. That's merely a refactoring PR.